resource-managers/kubernetes/core/src/test/scala/org/apache/spark/deploy/kubernetes/submit/PythonSubmissionResourcesSuite.scala

```diff
@@ -0,0 +1,133 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.deploy.kubernetes.submit
+
+import org.apache.spark.{SSLOptions, SparkConf, SparkFunSuite}
+import org.apache.spark.deploy.kubernetes.config._
+import scala.collection.JavaConverters._
+
+import io.fabric8.kubernetes.api.model.{ContainerBuilder, Pod, PodBuilder}
+
+
+
+private[spark] class PythonSubmissionResourcesSuite extends SparkFunSuite {
+  private val PYSPARK_FILES = Seq(
+    "hdfs://localhost:9000/app/files/file1.py",
+    "file:///app/files/file2.py",
+    "local:///app/files/file3.py",
+    "http://app/files/file4.py",
+    "file:///app/files/file5.py")
+  private val RESOLVED_PYSPARK_FILES = Seq(
+    "hdfs://localhost:9000/app/files/file1.py",
+    "/var/spark-data/spark-files/file2.py",
+    "local:///app/file`s/file3.py",
+    "http://app/files/file4.py")
+  private val PYSPARK_PRIMARY_FILE = "file:///app/files/file5.py"
+  private val RESOLVED_PYSPARK_PRIMARY_FILE = "/var/data/spark-files/file5.py"
+
+  private val pyFilesResource = new PythonSubmissionResourcesImpl(
+    PYSPARK_PRIMARY_FILE, Array(PYSPARK_FILES.mkString(","), "500")
+  )
+  private val pyResource = new PythonSubmissionResourcesImpl(
+    PYSPARK_PRIMARY_FILE, Array(null, "500")
+  )
+  private val SPARK_FILES = Seq.empty[String]
+  private val SPARK_JARS = Seq.empty[String]
+  private val JARS_DOWNLOAD_PATH = "/var/data/spark-jars"
+  private val FILES_DOWNLOAD_PATH = "/var/data/spark-files"
+  private val localizedFilesResolver = new ContainerLocalizedFilesResolverImpl(
```

Comment:
Generally a given test should only use the concrete implementation for the class that is under test. All other logical units should be mocks.